SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LocalReceptionistProxyServerManager.h"
+
+#include "Internationalization/Regex.h"
+#include "Sockets.h"
+#include "SocketSubsystem.h"
+#include "SpatialCommandUtils.h"
+#include "UObject/CoreNet.h"
+
+DEFINE_LOG_CATEGORY(LogLocalReceptionistProxyServerManager);
+
+#define LOCTEXT_NAMESPACE "FLocalReceptionistProxyServerManager"
+namespace
+{
+	static const int32 ExitCodeSuccess = 0;
+}
+
+FLocalReceptionistProxyServerManager::FLocalReceptionistProxyServerManager()
+: RunningCloudDeploymentName("")
+{
+}
+
+bool FLocalReceptionistProxyServerManager::GetProcessName(const FString& PID, FString& ProcessName)
+{
+	bool bSuccess = false;
+	ProcessName = TEXT("");
+	const FString TaskListCmd = TEXT("tasklist");
+
+	// get the task list line for the process with Pid 
+	const FString TaskListArgs = FString::Printf(TEXT(" /fi \"PID eq %s\" /nh /fo:csv"), *PID);
+	FString TaskListResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*TaskListCmd, *TaskListArgs, &ExitCode, &TaskListResult, &StdErr);
+	if (ExitCode == ExitCodeSuccess && bSuccess)
+	{
+		FRegexPattern ProcessNamePattern(TEXT("\"(.+?)\""));
+		FRegexMatcher ProcessNameMatcher(ProcessNamePattern, TaskListResult);
+		if (ProcessNameMatcher.FindNext())
+		{
+			ProcessName = ProcessNameMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+
+			return true;
+		}
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("%s"), *LOCTEXT("FailedToGetBlockingProcessName", "Failed to get the name of the process that is blocking the required port.").ToString());
+	
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::CheckIfPortIsBound(int32 Port, FString& PID, FText& LogMsg)
+{
+	const FString NetStatCmd = TEXT("netstat");
+
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString NetStatArgs = TEXT("-n -o -a");
```

Comment:
It seems a big portion of this is copied from `LocalDeploymentManager`, though I can't tell at a glance what we do differently here. Can we refactor these into helper functions in `SpatialCommandUtils` and have both the receptionist proxy server manager and local deployment manager use them?

Reply:
The part that we get the PID of the process can be a helper function at `SpatialCommandUtils` the rest are different. I will add it .